app/src/main/java/com/example/benchproject/app/movie/details/viewModel/MovieDetailsViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.benchproject.app.movie.details.viewModel
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.benchproject.app.movie.details.entity.MovieDetails
+import com.example.benchproject.domain.movie.details.entity.mapToUI
+import com.example.benchproject.domain.movie.details.usecase.GetMovieDetailsUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MovieDetailsViewModel @Inject constructor(
+    private val getMovieDetailsUseCase: GetMovieDetailsUseCase
+) : ViewModel() {
+
+    val responseMovieDetails: LiveData<MovieDetails>
+        get() = _responseMovieDetails
+
+    private val _responseMovieDetails = MutableLiveData<MovieDetails>()
+
+    fun getMovieDetails(movieId: Int) = viewModelScope.launch {
```

Comment:
What is the returned type here?

Reply:
Or this return nothing it's save data to live data or when it take result.failure from usecase it show message in log (in nearly PR #42 it will be show snackbar)

Reply:
I created [issue](https://github.com/Nataniel-Antosik/Bench_Project/issues/46) for this